diff --git a/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js b/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js
--- a/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js
+++ b/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js
@@ -88,7 +88,18 @@
   /**
    * See `NativeModeChangeEvent`.
    */
-  onModeChange?: ?DirectEventHandler<NativeModeChangeEvent>,
+  onModeChange?: ?DirectEventHandler<
+    Readonly<{
+      mode: Int32,
+      targetRectX: Double,
+      targetRectY: Double,
+      targetRectWidth: Double,
+      targetRectHeight: Double,
+      thresholdRectX: Double,
+      thresholdRectY: Double,
+      thresholdRectWidth: Double,
+      thresholdRectHeight: Double,
+    }>
+  >,
 }>;
 
 // TODO: Rename to eliminate "Experimental" suffix in the name.
diff --git a/src/private/components/virtualview/VirtualViewNativeComponent.js b/src/private/components/virtualview/VirtualViewNativeComponent.js
--- a/src/private/components/virtualview/VirtualViewNativeComponent.js
+++ b/src/private/components/virtualview/VirtualViewNativeComponent.js
@@ -88,7 +88,18 @@
   /**
    * See `NativeModeChangeEvent`.
    */
-  onModeChange?: ?DirectEventHandler<NativeModeChangeEvent>,
+  onModeChange?: ?DirectEventHandler<
+    Readonly<{
+      mode: Int32,
+      targetRectX: Double,
+      targetRectY: Double,
+      targetRectWidth: Double,
+      targetRectHeight: Double,
+      thresholdRectX: Double,
+      thresholdRectY: Double,
+      thresholdRectWidth: Double,
+      thresholdRectHeight: Double,
+    }>
+  >,
 }>;
 
 export default codegenNativeComponent<VirtualViewNativeProps>('VirtualView', {
