diff --git a/node_modules/react-native/src/private/components/virtualview/VirtualViewNativeComponent.js b/node_modules/react-native/src/private/components/virtualview/VirtualViewNativeComponent.js
--- a/node_modules/react-native/src/private/components/virtualview/VirtualViewNativeComponent.js
+++ b/node_modules/react-native/src/private/components/virtualview/VirtualViewNativeComponent.js
@@ -91,1 +91,1 @@
-  onModeChange?: ?DirectEventHandler<NativeModeChangeEvent>,
+  onModeChange?: DirectEventHandler<NativeModeChangeEvent>,
diff --git a/node_modules/react-native/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js b/node_modules/react-native/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js
--- a/node_modules/react-native/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js
+++ b/node_modules/react-native/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js
@@ -91,1 +91,1 @@
-  onModeChange?: ?DirectEventHandler<NativeModeChangeEvent>,
+  onModeChange?: DirectEventHandler<NativeModeChangeEvent>,
