diff --git a/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js b/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js
--- a/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js
+++ b/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js
@@ -85,10 +85,6 @@
 type VirtualViewNativeProps = Readonly<{
   ...ViewProps,
 
-  /**
-   * See `NativeModeChangeEvent`.
-   */
-  onModeChange?: ?DirectEventHandler<NativeModeChangeEvent>,
 }>;
 
 // TODO: Rename to eliminate "Experimental" suffix in the name.
diff --git a/src/private/components/virtualview/VirtualViewNativeComponent.js b/src/private/components/virtualview/VirtualViewNativeComponent.js
--- a/src/private/components/virtualview/VirtualViewNativeComponent.js
+++ b/src/private/components/virtualview/VirtualViewNativeComponent.js
@@ -85,10 +85,6 @@
 type VirtualViewNativeProps = Readonly<{
   ...ViewProps,
 
-  /**
-   * See `NativeModeChangeEvent`.
-   */
-  onModeChange?: ?DirectEventHandler<NativeModeChangeEvent>,
 }>;
 
 export default codegenNativeComponent<VirtualViewNativeProps>('VirtualView', {
