diff --git a/src/private/components/virtualview/VirtualViewNativeComponent.js b/src/private/components/virtualview/VirtualViewNativeComponent.js
--- a/src/private/components/virtualview/VirtualViewNativeComponent.js
+++ b/src/private/components/virtualview/VirtualViewNativeComponent.js
@@ -91,1 +91,1 @@
-  onModeChange?: ?DirectEventHandler<NativeModeChangeEvent>,
+  onModeChange?: DirectEventHandler<NativeModeChangeEvent>,
diff --git a/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js b/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js
--- a/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js
+++ b/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js
@@ -91,1 +91,1 @@
-  onModeChange?: ?DirectEventHandler<NativeModeChangeEvent>,
+  onModeChange?: DirectEventHandler<NativeModeChangeEvent>,
