diff --git a/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js b/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js
--- a/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js
+++ b/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js
@@ -88,7 +88,23 @@
   /**
    * See `NativeModeChangeEvent`.
    */
-  onModeChange?: ?DirectEventHandler<NativeModeChangeEvent>,
+  onModeChange?: DirectEventHandler<
+    Readonly<{
+      mode: Int32,
+      targetRect: Readonly<{
+        x: Double,
+        y: Double,
+        width: Double,
+        height: Double,
+      }>,
+      thresholdRect: Readonly<{
+        x: Double,
+        y: Double,
+        width: Double,
+        height: Double,
+      }>,
+    }>,
+  >,
 }>;
 
 // TODO: Rename to eliminate "Experimental" suffix in the name.
diff --git a/src/private/components/virtualview/VirtualViewNativeComponent.js b/src/private/components/virtualview/VirtualViewNativeComponent.js
--- a/src/private/components/virtualview/VirtualViewNativeComponent.js
+++ b/src/private/components/virtualview/VirtualViewNativeComponent.js
@@ -88,7 +88,23 @@
   /**
    * See `NativeModeChangeEvent`.
    */
-  onModeChange?: ?DirectEventHandler<NativeModeChangeEvent>,
+  onModeChange?: DirectEventHandler<
+    Readonly<{
+      mode: Int32,
+      targetRect: Readonly<{
+        x: Double,
+        y: Double,
+        width: Double,
+        height: Double,
+      }>,
+      thresholdRect: Readonly<{
+        x: Double,
+        y: Double,
+        width: Double,
+        height: Double,
+      }>,
+    }>,
+  >,
 }>;
 
 export default codegenNativeComponent<VirtualViewNativeProps>('VirtualView', {
